sdks/python/apache_beam/yaml/yaml_ml.py

```diff
@@ -33,11 +40,411 @@
+                location: us-east1
+                preprocess:
+                  callable: 'lambda x: {"prompt": x.prompt, "max_tokens": 50}'
+
+    Args:
+      endpoint_id: the numerical ID of the Vertex AI endpoint to query.
+      project: the GCP project name where the endpoint is deployed.
+      location: the GCP location where the endpoint is deployed.
+      experiment: Experiment label to apply to the
+        queries. See
+        https://cloud.google.com/vertex-ai/docs/experiments/intro-vertex-ai-experiments
+        for more information.
+      network: The full name of the Compute Engine
+        network the endpoint is deployed on; used for private
+        endpoints. The network or subnetwork Dataflow pipeline
+        option must be set and match this network for pipeline
+        execution.
+        Ex: "projects/12345/global/networks/myVPC"
+      private: If the deployed Vertex AI endpoint is
+        private, set to true. Requires a network to be provided
+        as well.
+      min_batch_size: The minimum batch size to use when batching
+        inputs.
+      max_batch_size: The maximum batch size to use when batching
+        inputs.
+      max_batch_duration_secs: The maximum amount of time to buffer
+        a batch before emitting; used in streaming contexts.
+      env_vars: Environment variables.
+      preprocess: A python callable, defined either inline, or using a file,
+        that is invoked on the input row before sending to the model to be
+        loaded by this ModelHandler. This parameter is required by the
+        `VertexAIModelHandlerJSON` ModelHandler.
+      postprocess: A python callable, defined either inline, or using a file,
+        that is invoked on the PredictionResult output by the ModelHandler
+        before parsing into the output Beam Row under the field name defined
+        by the inference_tag.
+    """
+
+    try:
+      from apache_beam.ml.inference.vertex_ai_inference import VertexAIModelHandlerJSON
+    except ImportError:
+      raise ValueError(
+          'Unable to import VertexAIModelHandlerJSON. Please '
+          'install gcp dependencies: `pip install apache_beam[gcp]`')
+
+    _handler = VertexAIModelHandlerJSON(
+        endpoint_id=str(endpoint_id),
+        project=project,
+        location=location,
+        experiment=experiment,
+        network=network,
+        private=private,
+        min_batch_size=min_batch_size,
+        max_batch_size=max_batch_size,
+        max_batch_duration_secs=max_batch_duration_secs,
+        env_vars=env_vars or {})
+
+    super().__init__(_handler, preprocess, postprocess)
+
+  @staticmethod
+  def validate(model_handler_spec):
+    pass
+
+  def inference_output_type(self):
+    return RowTypeConstraint.from_fields([('example', Any), ('inference', Any),
+                                          ('model_id', Optional[str])])
+
+  @staticmethod
+  def default_postprocess_fn():
+    return lambda x: beam.Row(**x._asdict())
```

Comment:
If x is a named tuple, we shouldn't have to convert it to a row to have nice schema'd output.